### PR TITLE
Refactor: Create base task suite and pipeline classes 

### PR DESCRIPTION
**Pull request by johnnygreco**
This PR reorganizes the task + pipeline framework. We now have `BaseTaskSuite` and `BasePipeline`, which provide the structure for creating pipelines for new use cases. 

A few other changes:
- added a `generate` method to `LLMWrapper`, which behaves like our inference api endpoints – i.e., it takes a string as input and returns a string, and it uses the same default system prompt.
- created a default llm config, which uses `gretelai-mistral-nemo-2407` for the nl, code, and judge models.
- updated the dev llm config in the examples folder to have the same default generation parameters as our inference api.

### CODE REVIEW

**Comment by johnnygreco** [comment]
@pimlock any objections for me to merge this?

File: navigator_helpers/llms/litellm.py [hunk 1]
```diff
@@ -43,6 +51,7 @@ def create_router(configs: list[LLMConfig]) -> Router:
 
 
 class LLMWrapper:
+    @staticmethod
```
Comment (pimlock) [inline navigator_helpers/llms/litellm.py:54]: Thx!

File: navigator_helpers/llms/gretel.py [hunk 2]
```diff
@@ -21,13 +23,16 @@ def completion(
         self,
         model: str,
         messages: list,
-        optional_params: dict[str, Any] = None,
+        optional_params: Optional[dict[str, Any]] = None,
```
Comment (pimlock) [inline navigator_helpers/llms/gretel.py:26]: Thanks for fixing this. I think that `=None` is not necessary and this seems to always be a dict ([code](https://github.com/BerriAI/litellm/blob/16c0307eab601b238f21021b9cba04b6957ad34c/litellm/llms/custom_llm.py#L94)), but we can leave it this way.

**Review by pimlock — APPROVED** [review]
Looks good! I reviewed the LLM changes, but need some more time to get through the pipeline stuff.

I wouldn't block on that at this point though, so 🚢 !

**Pull request merged**